### PR TITLE
Join game

### PR DESCRIPTION
**Pull request by Safisyx**
So, this is an update where a user can join a game.
I changed the button according to the current user :smile: 
lobby

To test it, make a little change in the Get('/games') to return all games not just the pending ones, or use the latest branch of the backend so that you test both at the same time :smile_cat: 

We still have this bug sometimes though, where we need to refresh the page to make it work :stuck_out_tongue:
bug

It mostly appears when a user create a new game, I think it's because of the socket which take some time to dispatch actions. We need to handle that delay, so I will be thinking about how to solve that before moving on.

### CODE REVIEW

File: src/components/game/GameDetails.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import './GameDetails.css'
+
+class GameDetails extends PureComponent {
+
+  handleClick = () => {
+    const {game,history,players} = this.props
+    switch(this.buttonText(players)) {
+      case "join":
+        this.props.joinGame(game.id)
+        history.push(`/games/${game.id}`)
```
Comment (Safisyx) [inline src/components/game/GameDetails.js:17]: This needs to be changed to go to the placing ships :smile:

File: src/components/game/GameDetails.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      default:
+        return null
+    }
+  }
+  buttonText = (players) => {
+    const id = this.props.userId
+    const game = this.props.game
+    if (players.includes(id)) {
+      if (game.status === "started") return "enter"
+      return "..."
```
Comment (Safisyx) [inline src/components/game/GameDetails.js:32]: I have an idea for this one, allow the player to leave the game if it's still pending. Then we delete that game! But we can leave that for later, add it to the todo list haha

Comment (Ju1111) [reply src/components/game/GameDetails.js:32]: so like leaving it without returning back to it?
yeah to do list haha 
did we actually write down those things somewhere? 😅

File: src/components/game/GameDetails.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  render() {
+    const {game,users,userId, players} = this.props
+    //const playerIds = game.players.map(player => player.userId)
+
+    if (game.status==="finished") return null
+
+    return (
+      <Card className="game-card">
+        <CardContent>
+          <Typography color="textSecondary">
```
Comment (Ju1111) [inline src/components/game/GameDetails.js:47]: what's the <Typography> tag? 
and the Cards XD ?

**Review by Ju1111 — APPROVED** [review]
Awesome! 
The thing with the delay is still weird yeah.. could we work with promises here? Or is that the problem.. that it goes on executing code while it waits at the same moment?
Also, should I commit my code of getting the actual boards from the backend? 
Will have to check if that still matches the backend code with the guess boards though 🤔

File: src/components/game/GameDetails.js (hunk 2 above)
Comment (Safisyx) [reply src/components/game/GameDetails.js:32]: Yes thing like that! 
Haha I though you have them somewhere :stuck_out_tongue: 
Well we can go through all reviews to find them :joy:

File: src/components/game/GameDetails.js (hunk 3 above)
Comment (Safisyx) [reply src/components/game/GameDetails.js:47]: I don't know, You put that right? haha. I just copied them :smile_cat: 
These are from material_ui but I'm not even sure which part of the render is that lol

**Comment by Safisyx** [comment]
I think I found the problem, when a new user signup, the list of users in the other connected users is not updated. So my idea is to emit an action to update those through the socket in the backend. I'm stilll working on it but later I will send a pull request for it in the backend.
About your code, maybe you only have to change little things before committing, I m not sure. But yeah, please make a pull request!

File: src/components/game/GameDetails.js (hunk 2 above)
Comment (Ju1111) [reply src/components/game/GameDetails.js:32]: I can try to make a list haha oops damn it

File: src/components/game/GameDetails.js (hunk 3 above)
Comment (Ju1111) [reply src/components/game/GameDetails.js:47]: oh omg of course the material_ui stuff. I already forgot how that works again.. weird stuff XD I think I do like CSS haha

**Pull request merged**